src/configure/model/templateModels.ts

```diff
@@ -13,18 +15,17 @@ export interface PipelineTemplate {
     language: string;
 }
 
-export interface PipelineTemplateMetadata {
+export interface TemplateInfo {
     templateId: string;
     workingDirectory: string;
     templateWeight: number;
     templateDescription: string;
     attributes: TemplateAttributes;
 }
 
-export interface RemotePipelineTemplate extends PipelineTemplate {
-    templateId: string;
+export interface RemotePipelineTemplate extends PipelineTemplate, ExtendedPipelineTemplate {
```

Comment:
Just thinking loud about this. I am very clearly able to segregate what does this model represent. It is having three things:
1. ExtendedpipelineTemplate properties
2. Local pipeline template fills - templateType, targetKind etc
3. Working directory which is already part of templateInfo

I understand that we want to handle both local and remote template as same object and hence this class extends PipelineTemplate. So, basically this class represents all properties of extendedPipelineTemplate as well as some properties from local template.
But what I don't completely understand is why workingDirectory should be part of this. Working directory is already part of TemplateInfo. can we use that object?

Reply:
TemplateInfo is not being used anywhere. When we get the list of templateInfo, we directly store it in  RemotePipelineTemplate object and use it afterwards that's why workingDirectory is part of it.